lib/uaa/token_coder.rb

```diff
@@ -70,7 +70,9 @@ def self.encode(token_body, options = {}, obsolete1 = nil, obsolete2 = nil)
     end
     options = normalize_options(options)
     algo = options[:algorithm]
-    segments = [Util.json_encode64("typ" => "JWT", "alg" => algo)]
+    header = {"typ" => "JWT", "alg" => algo}
+    header["kid"] = options.delete(:kid) if options[:kid]
+    segments = [Util.json_encode64(header)]
```

Comment:
It would be helpful for readers of the code if the `json_encode64()` set of methods were named to match what they actually do, ie they are a utility for producing non-padded URL safe base64 encoding (https://tools.ietf.org/html/rfc4648#section-5) - as distinct from the "standard" encoding which is called "base64" (https://tools.ietf.org/html/rfc4648#section-4).

The distinction is subtle - often (but not always) they will produce the same output, making it hard to detect an error if unlucky during testing (guess how I learnt this... :))

Reply:
I'll investigate renaming methods in another PR.